src/Service/Indexing/IndexingElasticService.php

```diff
@@ -0,0 +1,253 @@
+<?php
+
+namespace App\Service\Indexing;
+
+use App\Exception\SearchIndexException;
+use Elastic\Elasticsearch\Client;
+use Elastic\Elasticsearch\ClientBuilder;
+use Elastic\Elasticsearch\Exception\AuthenticationException;
+use Elastic\Elasticsearch\Exception\ClientResponseException;
+use Elastic\Elasticsearch\Exception\MissingParameterException;
+use Elastic\Elasticsearch\Exception\ServerResponseException;
+
+class IndexingElasticService implements IndexingServiceInterface
+{
+    private string $hostUrl;
+    private string $newIndexName;
+    private string $indexAliasName;
+
+    private Client $client;
+
+    public function __construct(string $bindIndexingUrl, string $bindIndexingAlias)
+    {
+        $this->hostUrl = $bindIndexingUrl;
+        $this->indexAliasName = $bindIndexingAlias;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add(IndexItem $item): void
+    {
+        /** @var IndexItem $item */
+        $params = [
+            'index' => $this->indexAliasName,
+            'id' => $item->getId(),
+            'body' => $item->toArray(),
+        ];
+
+        try {
+            $response = $this->getClient()->index($params);
+            $this->refreshIndex($this->indexAliasName);
+        } catch (SearchIndexException|ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode());
+        }
+
+        if (201 !== $response->getStatusCode()) {
+            throw new SearchIndexException('Unable to create new index', $response->getStatusCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function remove(int $id): void
+    {
+        $params = [
+            'index' => $this->indexAliasName,
+            'id' => $id,
+        ];
+
+        try {
+            $response = $this->getClient()->delete($params);
+            $this->refreshIndex($this->indexAliasName);
+        } catch (SearchIndexException|ClientResponseException|MissingParameterException|ServerResponseException $e) {
+            throw new SearchIndexException($e->getMessage(), (int) $e->getCode());
+        }
+
+        if (200 !== $response->getStatusCode()) {
+            throw new SearchIndexException('Unable to create new index', $response->getStatusCode());
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function bulkAdd(array $items): void
+    {
+        if (!isset($this->newIndexName)) {
+            $this->newIndexName = $this->indexAliasName.'_'.date('Y-m-d-His');
+            $this->createIndex($this->newIndexName);
+        }
+
+        $params = [];
+        foreach ($items as $item) {
```

Comment:
Should it be the responsibility of the service or of the caller to split this in sizeable chunks?  
https://www.elastic.co/guide/en/elasticsearch/reference/current/tune-for-indexing-speed.html#_use_bulk_requests

Reply:
As we talked about i think the caller as that is also us.